### PR TITLE
Add configurable cron expression example in scheduler guide

### LINKED ISSUE #5126: make Scheduler cron configurable via MP-Config
**Description**
I am using quarkus scheduler for a registration task for IoT devices in a smart city application. The task repeats every minute and retries to register if something went wrong before. It would be very helpful if I dont have to hardcode the cron into my services. Then i can repeat the task more often in dev mode and less often in production for example.

**Implementation ideas**
I think of a configuration key in the annotation like in microprofiles rest client `@RegisterRestClient(configKey = "...")`.
```
@Scheduled(configKey = "registrationcron")
public void register() {
    //...
}
```

### PR DESCRIPTION
**Pull request by machi1990**
fixes https://github.com/quarkusio/quarkus/issues/5126

/cc @gsmet @alexander-dammeier

### CODE REVIEW

File: docs/src/main/asciidoc/scheduler.adoc [hunk 1]
```diff
@@ -97,8 +107,17 @@ public class CounterBean {
 (Quarkus is using 10 worker threads for the scheduler). If it is not available the method invocation should be re-scheduled by default i.e
 it should be invoked as soon as possible. The invocation of the scheduled method does not depend on the status or result of the previous invocation.
 4. The code is pretty straightforward. Every 10 seconds, the counter is incremented.
+5. Define a job with a cron-like expressions. The syntax is currently based on Quartz Cron Trigger. The annotated method is executed at 10:15am every day.
```
Comment (gsmet) [inline docs/src/main/asciidoc/scheduler.adoc]:
```suggestion
5. Define a job with a cron-like expression. The syntax is currently based on Quartz Cron Trigger. The annotated method is executed at 10:15am every day.
```
Maybe we could add a link to the Quartz documentation?

File: docs/src/main/asciidoc/scheduler.adoc [hunk 2]
```diff
@@ -97,8 +107,17 @@ public class CounterBean {
...
+6. Define a job with a cron-like expressions `cron.expr` defined in `application.properties`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/scheduler.adoc]:
```suggestion
6. Define a job with a cron-like expression `cron.expr` which is configurable in `application.properties`.
```

File: docs/src/main/asciidoc/scheduler.adoc [hunk 3]
```diff
@@ -97,8 +107,17 @@ public class CounterBean {
...
+
+== Updating the application configuration file
+
+Edit the `application.properties` file and add the `cron.expr` configuration:
+[source,shell]
+----
+cron.expr=* * * * * ?
```
Comment (gsmet) [inline docs/src/main/asciidoc/scheduler.adoc]: Maybe make it every 5 seconds instead?

File: docs/src/main/asciidoc/scheduler.adoc [hunk 4]
```diff
@@ -89,6 +90,15 @@ public class CounterBean {
...
     }
 
+    @Scheduled(cron="0 15 10 * * ?") <5>
+    void cronJob(ScheduledExecution execution) {
+        System.out.println(execution.getScheduledFireTime());
+    }
+
+    @Scheduled(cron = "{cron.expr}") <6>
+    void cronJobWithExpressionInConfig() {
+       System.out.println("Cron expression from app config triggered");
```
Comment (gsmet) [inline docs/src/main/asciidoc/scheduler.adoc]:
```suggestion
       System.out.println("Cron expression configured in application.properties");
```

File: docs/src/main/asciidoc/scheduler.adoc [hunk 5]
```diff
@@ -166,5 +185,8 @@ Run the application with: `./mvnw compile quarkus:dev`.
 In another terminal, run `curl localhost:8080/count` to check the counter value.
 After a few seconds, re-run `curl localhost:8080/count` to verify the counter has been incremented.
 
+Observe the console to verify that the message `Cron expression from app config triggered` has been displayed indicating
```
Comment (gsmet) [inline docs/src/main/asciidoc/scheduler.adoc]:
```suggestion
Observe the console to verify that the message `Cron expression configured in application.properties` has been displayed indicating
```

File: docs/src/main/asciidoc/scheduler.adoc [hunk 6]
```diff
@@ -166,5 +185,8 @@ Run the application with: `./mvnw compile quarkus:dev`.
...
+that cron job with a configurable expression has been triggered.
```
Comment (gsmet) [inline docs/src/main/asciidoc/scheduler.adoc]:
```suggestion
that the cron job using an expression configured in `application.properties` has been triggered.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a few suggestions. Some of them will impact the quickstart.

File: docs/src/main/asciidoc/scheduler.adoc (hunk 1 above)
Comment (machi1990) [reply docs/src/main/asciidoc/scheduler.adoc]: Done

File: docs/src/main/asciidoc/scheduler.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/scheduler.adoc]: Done

File: docs/src/main/asciidoc/scheduler.adoc (hunk 3 above)
Comment (machi1990) [reply docs/src/main/asciidoc/scheduler.adoc]: Yeap, should be okay now.

**Comment by machi1990** [comment]
> I added a few suggestions. Some of them will impact the quickstart.

Thanks @gsmet  for the review. Suggestions incorporated.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**